hy/compiler.py

```diff
@@ -2017,19 +2017,24 @@ def compile_expression(self, expression):
 
             if fn.startswith("."):
                 # (.split "test test") -> "test test".split()
+                # (.a.b.c x) -> (.c (. x a b)) ->  x.a.b.c()
 
-                # Get the attribute name
-                ofn = fn
-                fn = HySymbol(ofn[1:])
-                fn.replace(ofn)
+                # Get the method name (the last named attribute
+                # in the chain of attributes)
+                attrs = [HySymbol(a).replace(fn) for a in fn.split(".")[1:]]
+                fn = attrs.pop()
 
-                # Get the object we want to take an attribute from
+                # Get the object we're calling the method on
+                # (extracted with the attribute access DSL)
                 if len(expression) < 2:
                     raise HyTypeError(expression,
                                       "attribute access requires object")
-                func = self.compile(expression.pop(1))
+                foo = HyExpression(
```

Comment:
Uhhh...`foo` is a really odd name here. What if you just put the whole thing in the call to `self.compile`? e.g.:

```
func = self.compile(HyExpression([HySymbol('.').replace(fn),
                                  expression.pop(1)] + attrs))
```